### PR TITLE
Add robots txt

### PR DESCRIPTION
**Pull request by m-wcislo**
**Description**

Adds robots.txt with sitemap. Allows indexing only /docs/latest/. According to indexing strategy longest rule match will apply, that is why robots.txt looks like:
```
User-agent: *
Allow: /docs/latest/
Disallow: /docs/
Sitemap: https://kyma-project.io/sitemap.xml
Host: https://kyma-project.io
```

**Related issue(s)**
Fixes: https://github.com/kyma-project/website/issues/382

### CODE REVIEW

File: gatsby/config-api/plugins.ts [hunk 1]
```diff
@@ -55,5 +55,11 @@ export const plugins: Plugins = [
       whitelist: ["GOOGLE_CSE", "ALGOLIA_API_KEY", "ALGOLIA_INDEX_NAME"],
     },
   },
+  {
+    resolve: "gatsby-plugin-robots-txt",
+    options: {
+      policy: [{ userAgent: "*", allow: "/docs/latest", disallow: "/docs/" }],
```
Comment (derberg) [inline gatsby/config-api/plugins.ts]: I think **allow** policy should end with `/`
so it should be `/docs/latest/`

comment based on examples visible here https://www.contentkingapp.com/academy/robotstxt/#order-of-precedence

File: README.md [hunk 2]
```diff
@@ -20,10 +20,11 @@ Use the following tools to set up the project:
 
 ### Install dependencies
 
-To install all dependencies, run this command:
+To install all dependencies, run those commands:
 
 ```
 npm install
+npm install --only=dev
```
Comment (magicmatatjahu) [inline README.md]: If you run `npm install`, also will install devDependencies, so this command in unnecessary.

File: gatsby/config-api/plugins.ts (hunk 1 above)
Comment (magicmatatjahu) [reply gatsby/config-api/plugins.ts]: We have redirect from `/docs/latest` to `/docs/latest/`, but please change `allow` as @derberg wrote.

Comment (m-wcislo) [reply gatsby/config-api/plugins.ts]: Yes, I did it due to redirect /docs/latest/ to /docs/latest to cover both. But anyway fixed.

File: README.md (hunk 2 above)
Comment (m-wcislo) [reply README.md]: Right, I tried to be too smart ;). Reverted lib fixing then.

**Pull request merged**